examples/llm/src/models/layers/gpt_blocks.py

```diff
@@ -33,17 +31,47 @@ def forward(self, x):
 
 class GPTBlock(nn.Module):
 
-    def __init__(self, cfg: DictConfig, device: Optional[str] = None):
+    def __init__(self,
+                 attn_impl: str,
+                 d_model: int,
+                 n_heads: int,
+                 mlp_ratio: int,
+                 attn_clip_qkv: Optional[float] = None,
+                 attn_qk_ln: bool = False,
+                 softmax_scale: Optional[float] = None,
+                 attn_pdrop: float = 0.0,
+                 alibi: bool = False,
+                 resid_pdrop: float = 0.0,
+                 low_precision_layernorm: bool = False,
+                 device: Optional[str] = None,
+                 **kwargs):
         super().__init__()
-        layernorm_class = LPLayerNorm if cfg.get('low_precision_layernorm',
-                                                 False) else nn.LayerNorm
+        if alibi:
+            assert attn_impl in {'triton', 'torch'
+                                }, 'Only triton kernel or torch supports alibi'
```

Comment:
isn't this already checked in MosaicGPTConfig?

Reply:
but also [hypocrisy](https://github.com/mosaicml/examples/pull/244#discussion_r1142698694)

Reply:
haha woops, i missed this one. yeah should be fine to check in the config as i do